src/compiler/checker.ts

```diff
@@ -21440,10 +21515,19 @@ namespace ts {
             }
             if (node.operand.kind === SyntaxKind.NumericLiteral) {
                 if (node.operator === SyntaxKind.MinusToken) {
-                    return getFreshTypeOfLiteralType(getLiteralType(-(<LiteralExpression>node.operand).text));
+                    const value = getNumericLiteralValue(node.operand as NumericLiteral);
+                    return getFreshTypeOfLiteralType(getLiteralType(
+                        // Keep number as number, bigint as bigint
+                        typeof value === "number"
+                            ? -value
+                            : { negative: !value.negative, base10Value: value.base10Value }
+                    ));
                 }

```

Comment:
In `checkGrammarNumericLiteral` you need to error if `options.experimentalBigInt` is not true and language is not `esnext` Note this is better place to error this than `getNumericLiteralValue`